alembic/config.py

```diff
@@ -103,8 +109,10 @@ def __init__(
         """Construct a new :class:`.Config`
 
         """
-        self.config_file_name = file_
```

Comment:
this is all very complicated, and won't work if someone specifies -c, why can't isnt ALEMBIC_CONFIG simply consumed inside of CommandLine.main() when we pass the name of the config file to the Config (one line change basically?)  is there some reason you need programmatic access to os.environ within the Config object itself  as well as the ability to make use of an alternate dictionary to os.environ ?